configure.py

```diff
@@ -2015,7 +2015,8 @@ def query_seastar_flags(pc_file, link_static_cxx=False):
                 f.write('build {}: cxx.{} {} || {}\n'.format(obj, mode, cc, ' '.join(serializers)))
                 if cc.endswith('Parser.cpp'):
                     # Unoptimized parsers end up using huge amounts of stack space and overflowing their stack
-                    flags = '-O1'
+                    flags = '-O1' if modes[mode]['optimization-level'] in ['0', 'g', 's'] else ''
+
```

Comment:
Looks good. Please file an issue. I normally don't like backporting performance patches, but this one can have such random effects (varying across point releases) that it's better to backport it.

Reply:
Filed https://github.com/scylladb/scylladb/issues/12463